client/api_bucket.go

```diff
@@ -309,15 +308,13 @@ func (c *client) PutBucketPolicy(ctx context.Context, bucketName string, princip
 }
 
 // DeleteBucketPolicy delete the bucket policy of the principal
-func (c *client) DeleteBucketPolicy(ctx context.Context, bucketName string, principalAddr string, opt types.DeletePolicyOption) (string, error) {
+func (c *client) DeleteBucketPolicy(ctx context.Context, bucketName string, principalStr types.Principal, opt types.DeletePolicyOption) (string, error) {
```

Comment:
Why not use the perms.Principal as the param directly?

It's a bit confusing do this type conversion.  perms.Principal ==> bytes ==> types.Principal(string)

Reply:
The user is unaware of this conversion process.  
The purpose of the design here is to consider that the user does not have chain knowledge, it is more convenient for him to use the sdk local interface.  At present, putObjectPolicy and putBucketPolicy is implemented in this way. I have communicated with test colleagues who have used these interfaces about the two ways of implementation. They are more willing to accept the current way of implementation.